### ISSUE TITLE
Failed to create network - possible concurrency issue?

### ISSUE THREAD

**Issue by lyricnz** [docker/docker#25393]
**Output of `docker version`:**

```
Client:
Version:      1.11.1
API version:  1.23
Go version:   go1.5.4
Git commit:   5604cbe
Built:        Tue Apr 26 23:30:23 2016
OS/Arch:      linux/amd64

Server:
Version:      1.12.0
API version:  1.24
Go version:   go1.6.3
Git commit:   8eab29e
Built:
OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 41
Running: 38
Paused: 0
Stopped: 3
Images: 437
Server Version: 1.12.0
Storage Driver: devicemapper
Pool Name: docker-253:0-204930528-pool
Pool Blocksize: 65.54 kB
Base Device Size: 10.74 GB
Backing Filesystem: xfs
Data file: /dev/dcspa14l_appvg/docker_data
Metadata file: /dev/dcspa14l_appvg/docker_metadata
Data Space Used: 116 GB
Data Space Total: 214.7 GB
Data Space Available: 98.76 GB
Metadata Space Used: 129 MB
Metadata Space Total: 5.369 GB
Metadata Space Available: 5.24 GB
Thin Pool Minimum Free Space: 21.47 GB
Udev Sync Supported: true
Deferred Removal Enabled: false
Deferred Deletion Enabled: false
Deferred Deleted Device Count: 0
Library Version: 1.02.107-RHEL7 (2015-10-14)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
Volume: local
Network: bridge null host overlay
Kernel Version: 3.10.0-327.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.2 (Maipo)
OSType: linux
Architecture: x86_64
CPUs: 24
Total Memory: 62.73 GiB
Name: dcspa14l.unix.anz
ID: FWD6:2DYT:2SPX:PJ4T:5NPF:NYX3:LJGE:5AMT:XOY3:ZBWU:N7EP:ZXVG
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Machine is physical hardware, AMD Opteron(tm) Processor 6174, 24 cores/cpu, 64GB RAM. Operating system is Redhat 7.2.  Dockerproject is only 3rd-party RPMs. Docker running natively on host.  /var/run/docker.sock exposed to (bamboo agent) containers via volume mount, actual "docker" commands running in the containers.  This works 100% fine on previous (VM) hosts, but now having occasional errors as described below.  

**Steps to reproduce the issue:**
1. System runs 24/7.  Multiple bamboo agents (each in docker containers) pick up jobs, and execute CI-type activities: building code etc, before spinning up an environment using docker-compose.  /var/run/docker.sock is volume-mapped from host to each guest.
2. A couple of times a day (maybe 1%) we get error below, and network fails to create
3. Each of the 5 bamboo agents should create their own separate network a1_default to a5_default

**Describe the results you received:**

```
03-Aug-2016 18:23:14 + set -e
03-Aug-2016 18:23:14 + docker network ls
03-Aug-2016 18:23:14 NETWORK ID          NAME                DRIVER
03-Aug-2016 18:23:14 99205c0d2a4d        agents_default      bridge             
03-Aug-2016 18:23:14 d538f34a8c1b        bridge              bridge             
03-Aug-2016 18:23:14 f1d1f3e1c863        host                host               
03-Aug-2016 18:23:14 a5a8158b87e3        none                null               
03-Aug-2016 18:23:14 + docker-compose down
03-Aug-2016 18:23:16 Removing network a4_default
03-Aug-2016 18:23:16 Network a4_default not found.
03-Aug-2016 18:23:16 + echo 'docker-compose down returned 0'
03-Aug-2016 18:23:16 + docker network ls
03-Aug-2016 18:23:16 docker-compose down returned 0
03-Aug-2016 18:23:16 NETWORK ID          NAME                DRIVER
03-Aug-2016 18:23:16 99205c0d2a4d        agents_default      bridge             
03-Aug-2016 18:23:16 d538f34a8c1b        bridge              bridge             
03-Aug-2016 18:23:16 f1d1f3e1c863        host                host               
03-Aug-2016 18:23:16 a5a8158b87e3        none                null               
03-Aug-2016 18:23:16 + sleep 5
03-Aug-2016 18:23:21 + docker-compose pull nginx simulator redis activemq oracle
03-Aug-2016 18:23:23 Pulling redis (csp-docker-registry.service.dev:5000/redis:latest)...
03-Aug-2016 18:23:23 latest: Pulling from redis
03-Aug-2016 18:23:23 Digest: sha256:eac6e2dec2f645d110b4013f1e6a59fa953a54d46fc5400172b7433fa0738593
03-Aug-2016 18:23:23 Status: Image is up to date for csp-docker-registry.service.dev:5000/redis:latest
03-Aug-2016 18:23:23 Pulling oracle (csp-docker-registry.service.dev:5000/oracle-12c:latest)...
03-Aug-2016 18:23:23 latest: Pulling from oracle-12c
03-Aug-2016 18:23:23 Digest: sha256:883722fd7a6c30c5f6c992104f0c29f4c9409737fef71eaac7f405000e75328e
03-Aug-2016 18:23:23 Status: Image is up to date for csp-docker-registry.service.dev:5000/oracle-12c:latest
03-Aug-2016 18:23:23 Pulling activemq (csp-docker-registry.service.dev:5000/rmohr/activemq:5.11.1)...
03-Aug-2016 18:23:23 5.11.1: Pulling from rmohr/activemq
03-Aug-2016 18:23:23 Digest: sha256:aebe8eeaf339ef4d5406b7881e87f939949a2311f69a7e4a11d4a21900d813db
03-Aug-2016 18:23:23 Status: Image is up to date for csp-docker-registry.service.dev:5000/rmohr/activemq:5.11.1
03-Aug-2016 18:23:23 Pulling simulator (csp-docker-registry.service.dev:5000/anz/simulator-develop:latest)...
03-Aug-2016 18:23:23 latest: Pulling from anz/simulator-develop
03-Aug-2016 18:23:45 Digest: sha256:55b64adf97f31c417888ba33b68d9e80d8f6bd4e6a9497ccc726add8fd6951ac
03-Aug-2016 18:23:45 Status: Downloaded newer image for csp-docker-registry.service.dev:5000/anz/simulator-develop:latest
03-Aug-2016 18:23:45 Pulling nginx (csp-docker-registry.service.dev:5000/nginx:latest)...
03-Aug-2016 18:23:45 latest: Pulling from nginx
03-Aug-2016 18:23:45 Digest: sha256:9671815a4cfa4d9b64ff808abe82b5cce895d9391511f3471270cb1a57753cf1
03-Aug-2016 18:23:45 Status: Image is up to date for csp-docker-registry.service.dev:5000/nginx:latest
03-Aug-2016 18:23:45 + docker-compose up -d --force-recreate nginx simulator redis activemq oracle
03-Aug-2016 18:23:47 Creating network "a4_default" with the default driver
03-Aug-2016 18:23:48 unable to remove jump to DOCKER-ISOLATION rule in FORWARD chain:  (iptables failed: iptables --wait -D FORWARD -j DOCKER-ISOLATION: iptables: No chain/target/match by that name.
03-Aug-2016 18:23:48 (exit status 1))
```

**Describe the results you expected:**
Network a4_default created per normal (and all other times works OK)

**Additional information you deem important (e.g. issue happens only occasionally):**

Main error is:

```
unable to remove jump to DOCKER-ISOLATION rule in FORWARD chain:  (iptables failed: iptables --wait -D FORWARD -j DOCKER-ISOLATION: iptables: No chain/target/match by that name.
```

Prior to 1.12.0 was getting errors like:

```
Jul 25 15:23:39 dcspa10l docker: time="2016-07-25T15:23:39.372045790+10:00" level=error msg="Handler for POST /v1.22/networks/create returned error: failed to parse pool request for address space \"LocalDefault\" pool \"\" subpool \"\": could not find an available predefined network"
```

Which looks a LOT like https://github.com/docker/libnetwork/issues/1101  !!   Haven't seen that since updated to 1.12.0, removed `/var/lib/docker/network/*` and rebooted.

**********\* Perhaps issue is concurrency?  There are FIVE bamboo agents, each of which could possibly be accessing docker at the same time.    ***********

Also noticed a race condition (??) between "docker-compose down" and subsequent "docker-compose up" - added delay.

Also had to disable network-manager completely, to stop lots of noise (??) with it noticing the interfaces.

Also lots of noise in dmesg like:

```
[271912.501730] XFS (dm-18): Mounting V4 Filesystem
[271912.598299] XFS (dm-18): Ending clean mount
[271912.599781] device veth866953f entered promiscuous mode
[271912.599897] IPv6: ADDRCONF(NETDEV_UP): veth866953f: link is not ready
[271912.599901] br-b5841fcebac4: port 4(veth866953f) entered forwarding state
[271912.599904] br-b5841fcebac4: port 4(veth866953f) entered forwarding state
[271912.600341] br-b5841fcebac4: port 4(veth866953f) entered disabled state
[271912.816148] IPv6: ADDRCONF(NETDEV_CHANGE): veth866953f: link becomes ready
[271912.816214] br-b5841fcebac4: port 4(veth866953f) entered forwarding state
[271912.816218] br-b5841fcebac4: port 4(veth866953f) entered forwarding state
[271913.095693] XFS (dm-19): Mounting V4 Filesystem
[271913.206525] XFS (dm-19): Ending clean mount
```

**Comment by lyricnz** [comment]
PS the network name a4_default is unique to each of the docker containers that could be running the "docker-compose up". The networks are a1-a5

**Comment by lyricnz** [comment]
PPS: correction on hardware details described above "Machine is physical hardware, Xeon E5-2660, 64GB RAM, 32 cores/cpu."  incorrect. This is actually "AMD Opteron(tm) Processor 6174, 24 cores/cpu, 64GB RAM".   

(4 of our CI machines are the Intel, 2 are the AMD - the particular incident cut and pasted above happened on the AMD)

**Comment by aboch** [comment]
@lyricnz 
I agree with your analysis, it is likely a concurrency issue where during network creation the bridge code first checks if the rule is present then it tries to remove it but in the meanwhile another network create request is doing the same and removes the rule first.
That code is not concurrent safe, needs to be fixed.

**Comment by aboch** [comment]
> Prior to 1.12.0 was getting errors like:
```
Jul 25 15:23:39 dcspa10l docker: time="2016-07-25T15:23:39.372045790+10:00" level=error msg="Handler for POST /v1.22/networks/create returned error: failed to parse pool request for address space \"LocalDefault\" pool \"\" subpool \"\": could not find an available predefined network"
```
> Which looks a LOT like docker/libnetwork#1101 [...]

Yes, that was fixed in 1.12.x via https://github.com/docker/libnetwork/pull/1130

**Comment by reembs** [comment]
Still getting this in version 1.12.4, build 1564f02

Our env is running the failing dockers concurrently (more than 4 starting at the same time) and inside a docker container (Ubuntu). Also happens rarely, probably less than 1% of times.

```
10:50:42 + docker-compose --verbose -f admin/docker/tx/system-docker-compose.yml up -d
10:50:43 docker.auth.find_config_file: Found file at path: /var/jenkins_home/.docker/config.json
10:50:43 docker.auth.load_config: Found 'auths' section
10:50:43 docker.auth.parse_auth: Found entry (registry=u'https://repo.host.com', username=u'AWS')
10:50:43 compose.cli.command.get_client: docker-compose version 1.10.0, build 4bd6f1a
10:50:43 docker-py version: 2.0.2
10:50:43 CPython version: 2.7.6
10:50:43 OpenSSL version: OpenSSL 1.0.1f 6 Jan 2014
10:50:43 compose.cli.command.get_client: Docker base_url: http+docker://localunixsocket
10:50:43 compose.cli.command.get_client: Docker version: KernelVersion=3.19.0-77-generic, Os=linux, BuildTime=2016-12-12T23:50:16.747144109+00:00, ApiVersion=1.24, Version=1.12.4, GitCommit=1564f02, Arch=amd64, GoVersion=go1.6.4
10:50:43 compose.cli.verbose_proxy.proxy_callable: docker info <- ()
10:50:44 compose.cli.verbose_proxy.proxy_callable: docker info -> {u'Architecture': u'x86_64',
10:50:44  u'BridgeNfIp6tables': True,
10:50:44  u'BridgeNfIptables': True,
10:50:44  u'CPUSet': True,
10:50:44  u'CPUShares': True,
10:50:44  u'CgroupDriver': u'cgroupfs',
10:50:44  u'ClusterAdvertise': u'',
10:50:44  u'ClusterStore': u'',
10:50:44  u'Containers': 4,
10:50:44  u'ContainersPaused': 0,
10:50:44 ...
10:50:44 compose.cli.verbose_proxy.proxy_callable: docker inspect_network <- (u'forterbranchname85systemtests_default')
10:50:44 compose.network.ensure: Creating network "forterbranchname85systemtests_default" with the default driver
10:50:44 compose.cli.verbose_proxy.proxy_callable: docker create_network <- (name=u'forterbranchname85systemtests_default', enable_ipv6=False, ipam=None, labels=None, driver=None, attachable=None, internal=False, options=None)
10:50:45 compose.cli.errors.log_api_error: unable to remove jump to DOCKER-ISOLATION rule in FORWARD chain:  (iptables failed: iptables --wait -D FORWARD -j DOCKER-ISOLATION: iptables: No chain/target/match by that name.
10:50:45  (exit status 1))
```

**Comment by b-a-t** [comment]
Getting the same with 1.13.1. We are creating docker networks via `systemd` and, apparently, it tries to do it in parallel, so at least one of the netwoks fails to create;
```
Feb 15 01:33:49 s3 dockerd[589]: time="2017-02-15T01:33:49.452141152+03:00" level=error msg="Handler for POST /v1.26/networks/create returned error: unable to remove jump to DOCKER-ISOLATION rule in FORWARD chain:  (iptables failed: iptables --wait -D FORWARD -j DOCKER-ISOLATION: iptables: No chain/target/match by that name.\n (exit status 1))"
Feb 15 01:33:49 s3 docker[759]: Error response from daemon: unable to remove jump to DOCKER-ISOLATION rule in FORWARD chain:  (iptables failed: iptables --wait -D FORWARD -j DOCKER-ISOLATION: iptables: No chain/target/match by that name.
Feb 15 01:33:49 s3 docker[759]: (exit status 1))
Feb 15 01:33:49 s3 systemd[1]: docker-network@production.service: main process exited, code=exited, status=1/FAILURE
Feb 15 01:33:49 s3 systemd[1]: Failed to start Production network.
```

**Comment by aboch** [comment]
@b-a-t  Thanks for confirming this  is a still reproducible in 1.13.x.
I have been trying to create/rm networks in parallel but was not able to repro yet.

I am working to a fix for this. If you could post your config file, it would help me validate the fix. Thanks.

**Comment by b-a-t** [comment]
As I said, we are using in-house written `systemd` unit files to roll out machine configuration. For networks we use a simple template unit file `docker-network@.service`:
```
[Unit]
Description=%I docker network service
Requires=docker.service
After=docker.service

[Service]
Type=oneshot
RemainAfterExit=on
ExecStart=/usr/bin/docker network create -d bridge %i
ExecStop=/usr/bin/docker network rm %i

[Install]
WantedBy=multi-user.target
```
and then create multiple instances by just putting into `/etc/systemd/system/docker-network@NETWORK.service.d/` simple unit file `unit.conf`:
```
[Unit]
Description=NETWORK network
```
Place instead of `NETWORK` appropriate name for the docker network, like `internal`, `gitlab`, `redmine`, etc.

Depending from the server configuration(and speed) this setup starts to fail with the 3 and/or more networks.

```
Feb 15 01:33:44 s3 dockerd[589]: time="2017-02-15T01:33:44.237849842+03:00" level=info msg="libcontainerd: new containerd process, pid: 600"
Feb 15 01:33:44 s3 kernel: [    9.283559] e1000e: eth1 NIC Link is Up 1000 Mbps Full Duplex, Flow Control: None
Feb 15 01:33:44 s3 kernel: [    9.283591] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
Feb 15 01:33:44 s3 kernel: [    9.283681] IPv6: ADDRCONF(NETDEV_CHANGE): eth1.10: link becomes ready
Feb 15 01:33:46 s3 dockerd[589]: time="2017-02-15T01:33:46.747516855+03:00" level=info msg="Graph migration to content-addressability took 0.00 seconds"
Feb 15 01:33:46 s3 dockerd[589]: time="2017-02-15T01:33:46.747701397+03:00" level=warning msg="Your kernel does not support cgroup rt period"
Feb 15 01:33:46 s3 dockerd[589]: time="2017-02-15T01:33:46.747711803+03:00" level=warning msg="Your kernel does not support cgroup rt runtime"
Feb 15 01:33:46 s3 dockerd[589]: time="2017-02-15T01:33:46.754567369+03:00" level=info msg="Loading containers: start."
Feb 15 01:33:46 s3 kernel: [   11.342613] bridge: automatic filtering via arp/ip/ip6tables has been deprecated. Update your scripts to load br_netfilter if you need thi
Feb 15 01:33:46 s3 kernel: [   11.345192] Bridge firewalling registered
Feb 15 01:33:46 s3 dockerd[589]: time="2017-02-15T01:33:46.912775376+03:00" level=info msg="Firewalld running: false"
Feb 15 01:33:46 s3 kernel: [   11.411137] ip_tables: (C) 2000-2006 Netfilter Core Team
Feb 15 01:33:47 s3 kernel: [   11.559961] Initializing XFRM netlink socket
Feb 15 01:33:47 s3 kernel: [   11.582960] IPv6: ADDRCONF(NETDEV_UP): docker_gwbridge: link is not ready
Feb 15 01:33:47 s3 kernel: [   11.600431] IPv6: ADDRCONF(NETDEV_UP): docker0: link is not ready
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.318327745+03:00" level=info msg="Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon option --bip can be used to set a preferred IP address"
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.409980143+03:00" level=info msg="Loading containers: done."
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.681867712+03:00" level=info msg="Listening for connections" addr="[::]:2377" proto=tcp
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.681867110+03:00" level=info msg="Listening for local connections" addr="/var/run/docker/swarm/control.sock" p
roto=unix
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.775296206+03:00" level=info msg="36315a9fafe8067c became follower at term 10"
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.775324045+03:00" level=info msg="newRaft 36315a9fafe8067c [peers: [], term: 10, commit: 113, applied: 0, last index: 113, lastterm: 10]"
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.815985633+03:00" level=info msg="36315a9fafe8067c is starting a new election at term 10"
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.816045405+03:00" level=info msg="36315a9fafe8067c became candidate at term 11"
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.816052890+03:00" level=info msg="36315a9fafe8067c received vote from 36315a9fafe8067c at term 11"
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.816063716+03:00" level=info msg="36315a9fafe8067c became leader at term 11"
Feb 15 01:33:47 s3 dockerd[589]: time="2017-02-15T01:33:47.816070478+03:00" level=info msg="raft.node: 36315a9fafe8067c elected leader 36315a9fafe8067c at term 11"
Feb 15 01:33:48 s3 dockerd[589]: time="2017-02-15T01:33:48.826445750+03:00" level=info msg="Initializing Libnetwork Agent Listen-Addr=0.0.0.0 Local-addr=10.21.30.4 Adv-addr=10.21.30.4 Remote-addr ="
Feb 15 01:33:48 s3 dockerd[589]: time="2017-02-15T01:33:48.826595084+03:00" level=info msg="Daemon has completed initialization"
Feb 15 01:33:48 s3 dockerd[589]: time="2017-02-15T01:33:48.826631383+03:00" level=info msg="Docker daemon" commit=092cba3 graphdriver=overlay2 version=1.13.1
Feb 15 01:33:48 s3 dockerd[589]: time="2017-02-15T01:33:48.826692947+03:00" level=info msg="Initializing Libnetwork Agent Listen-Addr=0.0.0.0 Local-addr=10.21.30.4 Adv-addr=10.21.30.4 Remote-addr ="
Feb 15 01:33:48 s3 dockerd[589]: time="2017-02-15T01:33:48.826756330+03:00" level=info msg="Gossip cluster hostname s3-5c4865ad2f78"
Feb 15 01:33:48 s3 dockerd[589]: time="2017-02-15T01:33:48.833828837+03:00" level=info msg="API listen on /var/run/docker.sock"
Feb 15 01:33:48 s3 systemd[1]: Started Docker Application Container Engine.
Feb 15 01:33:48 s3 systemd[1]: Starting Gitlab Pre-Production network...
Feb 15 01:33:48 s3 systemd[1]: Starting Internal network...
Feb 15 01:33:48 s3 systemd[1]: Starting Gitlab network...
Feb 15 01:33:49 s3 kernel: [   13.571277] br0: renamed from ov-001000-v5chi
Feb 15 01:33:49 s3 kernel: [   13.618066] vxlan1: renamed from vx-001000-v5chi
Feb 15 01:33:49 s3 kernel: [   13.638016] br0: port 1(vxlan1) entered blocking state
Feb 15 01:33:49 s3 kernel: [   13.638017] br0: port 1(vxlan1) entered disabled state
Feb 15 01:33:49 s3 kernel: [   13.638055] device vxlan1 entered promiscuous mode
Feb 15 01:33:49 s3 kernel: [   13.638186] br0: port 1(vxlan1) entered blocking state
Feb 15 01:33:49 s3 kernel: [   13.638187] br0: port 1(vxlan1) entered forwarding state
Feb 15 01:33:49 s3 kernel: [   13.726065] veth2: renamed from vethe5e5296
Feb 15 01:33:49 s3 kernel: [   13.753978] br0: port 2(veth2) entered blocking state
Feb 15 01:33:49 s3 kernel: [   13.753979] br0: port 2(veth2) entered disabled state
Feb 15 01:33:49 s3 kernel: [   13.754021] device veth2 entered promiscuous mode
Feb 15 01:33:49 s3 kernel: [   13.754069] br0: port 2(veth2) entered blocking state
Feb 15 01:33:49 s3 kernel: [   13.754069] br0: port 2(veth2) entered forwarding state
Feb 15 01:33:49 s3 dockerd[589]: time="2017-02-15T01:33:49+03:00" level=info msg="Firewalld running: false"
Feb 15 01:33:49 s3 kernel: [   13.850078] eth0: renamed from vethdd73dbe
Feb 15 01:33:49 s3 kernel: [   13.874507] IPv6: ADDRCONF(NETDEV_CHANGE): veth2: link becomes ready
Feb 15 01:33:49 s3 dockerd[589]: time="2017-02-15T01:33:49.452141152+03:00" level=error msg="Handler for POST /v1.26/networks/create returned error: unable to remove jump to DOCKER-ISOLATION rule in FORWARD chain:  (iptables failed: iptables --wait -D FORWARD -j DOCKER-ISOLATION: iptables: No chain/target/match by that name.\n (exit status 1))"
Feb 15 01:33:49 s3 docker[759]: Error response from daemon: unable to remove jump to DOCKER-ISOLATION rule in FORWARD chain:  (iptables failed: iptables --wait -D FORWARD -j DOCKER-ISOLATION: iptables: No chain/target/match by that name.
Feb 15 01:33:49 s3 docker[759]: (exit status 1))
Feb 15 01:33:49 s3 systemd[1]: docker-network@gitlab.service: main process exited, code=exited, status=1/FAILURE
Feb 15 01:33:49 s3 systemd[1]: Failed to start Gitlab network.
Feb 15 01:33:49 s3 systemd[1]: Dependency failed for postgresql/gitlab docker service.
Feb 15 01:33:49 s3 systemd[1]: Dependency failed for redis/gitlab docker service.
Feb 15 01:33:49 s3 systemd[1]: Dependency failed for gitlab docker service.
Feb 15 01:33:49 s3 systemd[1]: Unit docker-network@gitlab.service entered failed state.
Feb 15 01:33:49 s3 kernel: [   13.911211] IPv6: ADDRCONF(NETDEV_UP): br-115ea7c85f37: link is not ready
Feb 15 01:33:49 s3 kernel: [   13.912291] docker_gwbridge: port 1(veth45b7c54) entered blocking state
Feb 15 01:33:49 s3 kernel: [   13.912292] docker_gwbridge: port 1(veth45b7c54) entered disabled state
Feb 15 01:33:49 s3 kernel: [   13.912321] device veth45b7c54 entered promiscuous mode
Feb 15 01:33:49 s3 kernel: [   13.912359] IPv6: ADDRCONF(NETDEV_UP): veth45b7c54: link is not ready
Feb 15 01:33:49 s3 kernel: [   13.912367] docker_gwbridge: port 1(veth45b7c54) entered blocking state
Feb 15 01:33:49 s3 kernel: [   13.912368] docker_gwbridge: port 1(veth45b7c54) entered forwarding state
Feb 15 01:33:49 s3 kernel: [   13.912511] docker_gwbridge: port 1(veth45b7c54) entered disabled state
Feb 15 01:33:49 s3 docker[757]: 115ea7c85f3758de23b3c2967e6d5f23398e59e9ff538af069b113253a51a61a
Feb 15 01:33:49 s3 systemd[1]: Started Gitlab Pre-Production network.
Feb 15 01:33:49 s3 systemd[1]: Starting redis/gitlab/pp docker service...
Feb 15 01:33:49 s3 systemd[1]: Starting postgresql/gitlab/pp docker service...
Feb 15 01:33:49 s3 kernel: [   14.126153] eth1: renamed from vethcb78f67
Feb 15 01:33:49 s3 kernel: [   14.146377] IPv6: ADDRCONF(NETDEV_CHANGE): veth45b7c54: link becomes ready
Feb 15 01:33:49 s3 kernel: [   14.146398] docker_gwbridge: port 1(veth45b7c54) entered blocking state
Feb 15 01:33:49 s3 kernel: [   14.146399] docker_gwbridge: port 1(veth45b7c54) entered forwarding state
Feb 15 01:33:49 s3 kernel: [   14.146416] IPv6: ADDRCONF(NETDEV_CHANGE): docker_gwbridge: link becomes ready
Feb 15 01:33:50 s3 kernel: [   14.634119] IPv6: eth1: IPv6 duplicate address fe80::42:acff:fe14:2 detected!
Feb 15 01:33:50 s3 kernel: [   14.939982] IPv6: ADDRCONF(NETDEV_UP): br-e9ff580acd21: link is not ready
Feb 15 01:33:50 s3 docker[758]: e9ff580acd21f698ec33852eaa49b3064941e2d34a9ea9fe6c4ee20f6538ca29
Feb 15 01:33:50 s3 systemd[1]: Started Internal network.
```

**Comment by b-a-t** [comment]
By adding yet another, dummy network I managed to start all networks without problems. Seems, it's all depends on timing and some other subtle factors.
```
Feb 15 04:56:07 s3 systemd[1]: Started Docker Application Container Engine.
Feb 15 04:56:07 s3 systemd[1]: Starting Zero network...
Feb 15 04:56:07 s3 systemd[1]: Starting Gitlab Pre-Production network...
Feb 15 04:56:07 s3 systemd[1]: Starting Internal network...
Feb 15 04:56:07 s3 systemd[1]: Starting Gitlab network...
```

**Comment by lyricnz** [comment]
Which configuration file?  I no longer work on the project that had the original problem, but may still have the file.

FWIW, I was using multiple parallel "docker-compose up" with ~5 services defined in each docker-compose.yml file, if that helps.

**Comment by b-a-t** [comment]
By adding yet another, dummy network I managed to start all networks without problems. Seems, it's all depends on timing and some other subtle factors.
```
Feb 15 04:56:07 s3 systemd[1]: Started Docker Application Container Engine.
Feb 15 04:56:07 s3 systemd[1]: Starting Zero network...
Feb 15 04:56:07 s3 systemd[1]: Starting Gitlab Pre-Production network...
Feb 15 04:56:07 s3 systemd[1]: Starting Internal network...
Feb 15 04:56:07 s3 systemd[1]: Starting Gitlab network...
```

**Comment by rodrigodiez** [comment]
Same problem here on docker 1.13.1, running 6 concurrent docker-compose up in a Jenkins machine usually ends up in the error stated above

**Comment by ebuildy** [comment]
Hello I see this ticket closed, but it's still happening right? thanks you

**Comment by aboch** [comment]
@ebuildy Issues get automatically closed when the fix is being merged to master.
The fix makes it to the release marked in the milestone.
So in this case the fix for this issue will be available in docker 17.04.

**Comment by ebuildy** [comment]
Thanks you, very good news for our CI team
